docs/03-computation/index.mdx

```diff
@@ -6,13 +6,61 @@ tags: ["introduction"]
 import DocCardList from '@theme/DocCardList';
 import {useCurrentSidebarCategory} from '@docusaurus/theme-common';
 
-The blindnet devkit **Privacy Computation Engine** is a service that interpret your rights to hold and treat a particular Data Capture at a particular point of time and calculate a response to [Data Subjects'](/docs/references/lexicon#data-subject) [Privacy Requests](/docs/references/lexicon#privacy-request).
+The blindnet devkit **Privacy Computation Engine** (PCE) is the core of your "privacy stack".
 
-It associates:
+It is a service delivering Restful APIs to manage the two core features of the DevKit by:
 
-- a [Privacy Compiler](/docs/references/high-level-architecture#privacy-compiler)
-- and [Customization API](/docs/references/high-level-architecture#customization-api), for you to set up your organization-specific information and parameters
+1. interpreting your rights to hold and treat a particular **[Data Capture](/docs/references/lexicon#data-capture)** at a particular point in time
+2. calculating a response to [Data Subjects'](/docs/references/lexicon#data-subject) **[Privacy Requests](/docs/references/lexicon#privacy-request)**.
 
----
+Both these features are provided by a main internal component we call the [Privacy Compiler](/docs/references/high-level-architecture#privacy-compiler), based on the specific information and parameters of your organization you define using the [Customization API](/docs/references/high-level-architecture#customization-api).
+
+## How It Works
+
+The Privacy Compiler inside the Privacy Computation Engine is here to deliver recommended actions based on a timeline of _Events_[^1] it proceeds.
+
+:::info key point
+
+The Privacy Compiler ***compiles*** a timeline of ***events*** into recommended actions.
+
+:::
+
+### Events
+
+Events are defined by:
+
+- their association with one related [Data Subject](/docs/references/lexicon#data-subject)
+- information needed to determine consequent actions
+- the specific date and time at which the event occurred
+
+As such, events can represent:
+
+- the capture of some sensitive data ([_Data Captures Fragments_](https://github.com/blindnet-io/product-management/blob/main/refs/schemas/priv/RFC-PRIV.md#data-capture-fragments))
+- an assertion of privacy rights from people about whom you hold sensitive information ([_Privacy Request_](https://github.com/blindnet-io/product-management/blob/main/refs/schemas/priv/RFC-PRIV.md#privacy-request))
```

Comment:
I didn't model _Privacy Requests_ as events, only the _Privacy Request Response_, because only GRANTED responses are relevant for calculating consequent actions (DELETE).

We can keep _Privacy Requests_ as events just conceptually in the docs (and even display them together with other events in the UI).

Reply:
I based this on the definition of events in the [PRIV RFC](https://github.com/blindnet-io/product-management/blob/main/refs/schemas/priv/RFC-PRIV.md#events).

As PCE is pretty central, the "computation" section of the documentation can cover topics with a way larger scope than the service itself. I might indeed need to make this clearer by removing references to the PCE in these docs' titles.

It seems indeed Privacy Requests should still be considered "events" conceptually. System-wise, a Privacy Request is an event, triggered by a Data Subject (i.e. a user event). But I understand the PCE itself can have a different usage of the word "event" internally, and a simple call to an API endpoint isn't considered an "event" per se.

Your comment makes me think we should further discuss how events are modelized in the whole system, and how we use this term everywhere (docs, specs, PRIV, etc.). Could you please give more details about events management in the PCE, specifically what you precisely think about when saying "I didn't model Privacy Requests as events"?

Reply:
In PCE, I considered PRIV events:
- legal base events (`RELEATIONSHIP/SERVICE-START/END` for `CONTRACT`, `NECESSARY` and `LEGITIMATE-INTEREST`)
- given `CONSENTS`
- responses for `REVOKE-CONSENT` demands
- responses for `OBJECT` and `RESTRICT` demands

With regards to regulations and retention policies, these events are used to calculate eligible privacy scope and (not yet implemented) automatic `DELETE` actions.

Of course, timeline can include more data just these events. Everything that has a `datetime` property and an associated user can be included in the timeline.

Reply:
Could you please point me to the associated code / PRs / issues?

Reply:
[Code](https://github.com/blindnet-io/privacy-computation-engine/blob/develop/modules/priv/src/main/scala/TimelineEvent.scala#L10)

Reply:
> We can keep _Privacy Requests_ as events just conceptually in the docs (and even display them together with other events in the UI).

Everyting that connects a date to a data subject is an Event. Now I understand that as you @m4rk055 implemented it, you did not explicitely store Privacy Requests at the same place (timeline) where you store the other events. This is a legitimate implementation decision, changing nothing to the nature of what things are.

We also want, at a later version, to be able to show the timeline to the DPO (data cosnumer) and allow them to use it to demonstrate compliance. Privacy Requests belong naturally in that timeline view along with responses, and other events.